src/solver/types.jl

```diff
@@ -468,45 +468,100 @@ function (cb::AtmosCallback)(integrator)
+function maybe_use_rpath(file)
+    rfile = relpath(file, dirname(config_path))
+    return if isfile(rfile) && samefile(rfile, file)
+        rfile
+    else
+        file
+    end
 end
 
 """
-    AtmosConfig(; comms_ctx = nothing)
-Helper function for the AtmosConfig constructor.
-Reads the `config_file` from the command line into a Dict
+    AtmosConfig(
+        config_file::String = default_config_file;
+        job_id = config_id_from_config_file(config_file),
+        comms_ctx = nothing,
+    )
+    AtmosConfig(
+        config_files::Union{NTuple{<:Any, String} ,Vector{String}};
+        job_id = config_id_from_config_files(config_files),
+        comms_ctx = nothing,
+    )
+
+Helper function for the AtmosConfig constructor. Reads a YAML file into a Dict
 and passes it to the AtmosConfig constructor.
 """
-function AtmosConfig(; comms_ctx = nothing)
-    parsed_args = parse_commandline(argparse_settings())
-    return AtmosConfig(parsed_args["config_file"]; comms_ctx)
-end
+AtmosConfig(
+    config_file::String = default_config_file;
+    job_id = config_id_from_config_file(config_file),
+    comms_ctx = nothing,
+) = AtmosConfig((config_file,); job_id, comms_ctx)
+
+function AtmosConfig(
+    config_files::ContainerType{String};
+    job_id = config_id_from_config_files(config_files),
+    comms_ctx = nothing,
+)
+
+    all_config_files =
+        if any(cf -> samefile(cf, default_config_file), config_files)
+            maybe_use_rpath.(config_files)
+        else
+            (
+                maybe_use_rpath(default_config_file),
+                maybe_use_rpath.(config_files)...,
+            )
+        end
 
-AtmosConfig(::Nothing; comms_ctx = nothing) = AtmosConfig(Dict(); comms_ctx)
+    configs = map(all_config_files) do config_file
+        @info "Loading yaml file $config_file"
+        strip_help_messages(YAML.load_file(config_file))
+    end
+    return AtmosConfig(
+        configs;
+        comms_ctx,
+        config_files = all_config_files,
+        job_id,
+    )
+end
 
 """
-    AtmosConfig(config::Dict; comms_ctx = nothing)
-Constructs the AtmosConfig from the Dict passed in. This Dict overrides all of
+    AtmosConfig(
+        configs::Union{NTuple{<:Any, Dict} ,Vector{Dict}};
+        comms_ctx = nothing,
+        config_files,
+        job_id
+    )
+
+Constructs the AtmosConfig from the Dicts passed in. This Dict overrides all of
 the default configurations set in `default_config_dict()`.
 """
-function AtmosConfig(config::Dict; comms_ctx = nothing)
-    config = override_default_config(config)
+AtmosConfig(configs::AbstractDict; kwargs...) =
+    AtmosConfig((configs,); kwargs...)
+function AtmosConfig(
+    configs::ContainerType{AbstractDict};
+    comms_ctx = nothing,
+    config_files = [default_config_file],
+    job_id = "",
+)
+    # using config_files = [default_config_file] as a default
+    # relies on the fact that override_default_config uses
+    # default_config_file.
```

Comment:
```suggestion
    # default_config_file.
    config_files = normpath.(config_files)
```
For nicer printing:
```julia
julia> CA.AtmosConfig(Dict())
...
┌ Info: Making AtmosConfig with config files: 
└    /Users/nefrathenrici/clima/ClimaAtmos.jl/config/default_configs/default_config.yml
```
instead of 
```julia
julia> CA.AtmosConfig(Dict())
...
┌ Info: Making AtmosConfig with config files: 
└    /Users/nefrathenrici/clima/ClimaAtmos.jl/src/solver/../../config/default_configs/default_config.yml
```

Reply:
Ah, I kind of did that in the other constructor. I'll add it to both.